go.mod

```diff
@@ -35,7 +35,7 @@ require (
 	github.com/golang/protobuf v1.5.4
 	github.com/golang/snappy v0.0.5-0.20231225225746-43d5d4cd4e0e
 	github.com/google/btree v1.0.1
-	github.com/google/pprof v0.0.0-20210827144239-02619b876842
+	github.com/google/pprof v0.0.0-20240227163752-401108e1b7e7
```

Comment:
Are these dependency upgrades intentional? They don't seem related to the stated purpose of this PR. Can they be separated into a different PR if so?

Reply:
Seems this is related to the fgprof change which doesn't seem related to the purpose of reducing roachprod startup lag?

Reply:
The fgprof change is how I diagnosed the roachprod startup lag. In the interest of keeping this PR smaller, I just removed the fgprof commit. There are no dep changes any more.